opentelemetry/proto/trace/v1/trace.proto

```diff
@@ -82,20 +82,24 @@ message Span {
   // the same `trace_id`. The ID is a 16-byte array. An ID with all zeroes
   // is considered invalid.
   //
-  // This field is semantically required. Receiver should generate new
-  // random trace_id if empty or invalid trace_id was received.
-  //
   // This field is required.
+  //
+  // The receivers SHOULD generate a new random trace_id if any of the following is true:
```

Comment:
Why use a random ID? Why not reject the data as invalid?

Reply:
That's the way it is currently. I can't change the current behavior, it would be a breaking change.
Why it was specified that way originally, I don't know.

Reply:
Where is this specified? It's specified for OTLP transport as well as OTLP/JSON transport? Thanks.

Reply:
> current behavior, it would be a breaking

Current behavior is broken, which I think allows us to change this behavior. How could anyone know the difference between a randomly-generated ID and an invalid data?

Reply:
This is currently specified in this repo: https://github.com/open-telemetry/opentelemetry-proto/blob/d8729d40f629dba12694b44c4c32c1eab109b00a/opentelemetry/proto/trace/v1/trace.proto#L85

Quote:
> Receiver should generate new random trace_id if empty or invalid trace_id was received.

Again, the aim of this PR is not to change any existing behavior, merely to clarify it.

If we believe what is currently specified is a bug that needs to be fixed then that's a different story. That was not the goal of my PR.

I agree with you that generating a random id when an invalid one is received doesn't sounds right. However, I don't want to change this without understanding why it was done so in the first place (and also because I think it would be a breaking change).

Reply:
I think this should definitely treat the whole span as invalid if the trace_id is invalid.

Reply:
So what to do with invalid spans?

Reply:
https://github.com/open-telemetry/opentelemetry-proto/pull/444 is now merged and we no longer generate random ids.